<p>This rule detects lack of hystrix fallback</p>
<h2>Noncompliant Code Example</h2>
<pre>
    Subclass of HystrixCommand or spring HystrixAnnotation without fallback
</pre>
<h2>Compliant Solution</h2>
<pre>
    All HystrixCommand need its fallback
</pre>