<p>This rule detects lack of hystrix fallback</p>
<h2>Noncompliant Code Example</h2>
<pre>
    <body>
        1. Hystrix subclass
        public class HelloWorldCommand extends HystrixCommand&lt;String&gt; {
            public HelloWorldCommand(String name) {
                super(Setter.withGroupKey(HystrixCommandGroupKey.Factory.asKey("HelloWorldGroup"))
                    .andCommandPropertiesDefaults(HystrixCommandProperties.Setter().withExecutionIsolationThreadTimeoutInMilliseconds(500)));
                this.name = name;
            }
        }

        2. Spring Annotation
        public class HelloService {
            @Autowired
            RestTemplate restTemplate;

            @HystrixCommand()
            public String hiService(String name) {
                return restTemplate.getForObject("http://SERVICE-HI/hi?name="+name,String.class);
            }

            public String hiError(String name) {
                return "hi,"+name+",sorry,error!";
            }
        }

        3. Spring Feign
        @FeignClient(value = "service-hi")
        public interface SchedualServiceHi {
            @RequestMapping(value = "/hi",method = RequestMethod.GET)
            String sayHiFromClientOne(@RequestParam(value = "name") String name);
        }
    </body>>
</pre>
<h2>Compliant Solution</h2>
<pre>
    <body>
        All HystrixCommand need its fallback
        1. Hystrix subclass
        public class HelloWorldCommand extends HystrixCommand&lt;String&gt; {
            public HelloWorldCommand(String name) {
                super(Setter.withGroupKey(HystrixCommandGroupKey.Factory.asKey("HelloWorldGroup"))
                    .andCommandPropertiesDefaults(HystrixCommandProperties.Setter().withExecutionIsolationThreadTimeoutInMilliseconds(500)));
                this.name = name;
            }

            @Override
            protected String getFallback() {
                return "exeucute Falled";
            }
        }

        2. Spring Annotation
        public class HelloService {
            @Autowired
            RestTemplate restTemplate;

            @HystrixCommand(fallbackMethod = "hiError")
            public String hiService(String name) {
                return restTemplate.getForObject("http://SERVICE-HI/hi?name="+name,String.class);
            }

            public String hiError(String name) {
                return "hi,"+name+",sorry,error!";
            }
        }

        3. Spring Feign
        @FeignClient(value = "service-hi",fallback = SchedualServiceHiHystric.class)
        public interface SchedualServiceHi {
            @RequestMapping(value = "/hi",method = RequestMethod.GET)
            String sayHiFromClientOne(@RequestParam(value = "name") String name);
        }
    </body>
</pre>